source/fundamentals/authentication.txt

```diff
@@ -1 +1,208 @@
+- ``hostname`` - The network address of your MongoDB server, accessible by your client
+- ``port`` - The port number of your MongoDB server
+- ``authenticationDb`` - The MongoDB database that contains your user's authentication
+  data. If you omit this parameter, the driver uses the default value ``admin``.
+
+Select the :guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax for specifying your authentication mechanism.
+
+.. _csharp-authentication-default:
+
+Default
+~~~~~~~
+
+The default authentication mechanism setting uses one of the following
+authentication mechanisms, depending on what MongoDB versions your server supports:
+
+- ``SCRAM-SHA-256``
+- ``SCRAM-SHA-1``
+- ``MONGODB-CR``
+
+.. note::
+
+   MongoDB version 4.0 uses SCRAM as the default mechanism, and no longer
+   supports ``MONGODB-CR``. These versions automatically determine which
+   version of SCRAM to use.
+
+
+Select the :guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax for specifying the default authentication mechanism:
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: default-connection-string
+
+      .. code-block:: csharp
+
+         var mongoClient = new MongoClient("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>");
+
+   .. tab:: MongoCredential
+      :tabid: default-mongo-credential
+
+      .. code-block:: csharp
+
+         var credentials = MongoCredential.CreateCredential("<authenticationDb>", "<username>", "<password>");
+         var settings = MongoClientSettings.FromConnectionString("<host>");
+         settings.Credential = credentials;
+         var mongoClient = new MongoClient(settings);
+
+.. _csharp-scram-sha-256:
+
+SCRAM-SHA-256
+~~~~~~~~~~~~~
+
+``SCRAM-SHA-256`` is a salted challenge-response authentication mechanism (SCRAM)
+that uses your username and password, encrypted with the ``SHA-256`` algorithm,
+to authenticate your user. 
+
+.. tip:: Default Mechanism
+
+   MongoDB version 4.0 and later uses SCRAM as the default authentication mechanism.
+   To learn more on specifying the default mechanism, see :ref:`csharp-authentication-default`.
+
+.. _csharp-scram-sha-1:
+
+SCRAM-SHA-1
+~~~~~~~~~~~
+
+``SCRAM-SHA-1`` is s a salted challenge-response mechanism (SCRAM) that uses 
+your username and password, encrypted with the ``SHA-1`` algorithm, to authenticate 
+your user.
+
+.. tip:: Default Mechanism
+
+   MongoDB version 4.0 and later uses SCRAM as the default authentication mechanism.
+   To learn more on specifying the default mechanism, see :ref:`csharp-authentication-default`.
+
+.. _csharp-x509:
+
+X.509
+~~~~~
+
+The ``X.509`` authentication mechanism uses `TLS <https://en.wikipedia.org/wiki/Transport_Layer_Security>`__
+with ``X.509`` certificates to authenticate your user, identified by the
+distinguished names of your client certificate. When you specify the 
```

Comment:
what does "distinguished names of your client certificate" mean? Or should we expect readers to understand?

Reply:
I think defining that is out of scope for this page. If the user is planning on using X509 I think they would have an idea on what this means, and they can read more information on it in the X.509 guide linked below.